### PR TITLE
Add categories to each changelog release

### PR DESCRIPTION
**Pull request by Gudahtt**
Each changelog release now has category headers. The standard ["keep a changelog"](https://keepachangelog.com/en/1.0.0/) categories are used, along with the addition of "Uncategorized" for any changes that have not yet been categorized.

The changelog script has been updated to add this "Uncategorized" header if it isn't already present, and to place any new commits under this header.

The changelog has been updated to property categorize each change in recent releases, and to place changes in older releases under the header "Uncategorized".

Relates to #10752

Manual testing steps:  

For development changelog updates:
- Run `yarn update-changelog`
- See that it places changelog entries under 'Unreleased`, under the "Uncategorized" header

For RC changelog updates with existing release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Add a new release header to the changelog for the new version
- Run `yarn update-changelog`
- See that it places changelog entries under the new release header under the "Uncategorized" category
 
For updates without a release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Run `yarn update-changelog`
- See that it places changelog entries under the newly added release header under the "Uncategorized" category

### CODE REVIEW

File: development/auto-changelog.js [hunk 1]
```diff
@@ -69,6 +69,37 @@ async function main() {
   const changelog = await fs.readFile(changelogFilename, { encoding: 'utf8' });
   const changelogLines = changelog.split('\n');
 
+  const prNumbersWithChangelogEntries = [];
```
Comment (Gudahtt) [inline development/auto-changelog.js:72]: This was moved earlier in the script to allow the early exit on line 99, to ensure we don't add the 'Uncategorized' header to the "Unreleased" section if there are no changes.

As a bonus, it now avoids creating a new release header if there are no commits as well! Though that shouldn't happen generally.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1844,42 +1861,40 @@ rollback to 3.10.0 due to bug
...
 - Fix a bug with estimating gas on Parity
 - Show loading indication when selecting ShapeShift as purchasing method.
 
 ## [2.13.0] - 2016-09-18
-
+### Uncategorized
 - Add Parity compatibility, fixing Geth dependency issues.
-- Add a link to the transaction in history that goes to https://metamask.github.io/eth-tx-viz
-  too help visualize transactions and to where they are going.
+- Add a link to the transaction in history that goes to https://metamask.github.io/eth-tx-viz to help visualize transactions and to where they are going.
```
Comment (Gudahtt) [inline CHANGELOG.md:1878]: I moved this onto one line to make the formatting more consistent and easier to parse. I also fixed a typo here (`too` => `to`).

File: CHANGELOG.md [hunk 3]
```diff
@@ -1844,42 +1861,40 @@ rollback to 3.10.0 due to bug
...
 - Show "Buy Ether" button and warning on tx confirmation when sender balance is insufficient
 
 ## [2.12.1] - 2016-09-14
-
-- Fixed bug where if you send a transaction from within MetaMask extension the
-  popup notification opens up.
+### Uncategorized
+- Fixed bug where if you send a transaction from within MetaMask extension the popup notification opens up.
```
Comment (Gudahtt) [inline CHANGELOG.md:1883]: This was moved onto one line as well.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2149,18 +2164,18 @@ rollback to 3.10.0 due to bug
...
+### Uncategorized
 - When unlocking, first account is auto-selected.
 - When creating a first vault on the test-net, the first account is auto-funded.
 - Fixed some styling issues.
 
 ## [1.0.0] - 2016-03-25
-
-Made seed word restoring BIP44 compatible.
+### Uncategorized
+- Made seed word restoring BIP44 compatible.
```
Comment (Gudahtt) [inline CHANGELOG.md:2174]: These older releases didn't have a `- ` in front of their changes. I've added it to make them consistent with the others.

**Review by brad-decker — DISMISSED** [review]
This LGTM though I'd like some clarification on how the various categories are determined. What differentiates changed/added vs fixed?

**Comment by Gudahtt** [comment]
The k[eep a changelog spec](https://keepachangelog.com/en/1.0.0/) describes the categories as follows:

- `Added` for new features.
- `Changed` for changes in existing functionality.
- `Deprecated` for soon-to-be removed features.
- `Removed` for now removed features.
- `Fixed` for any bug fixes.
- `Security` in case of vulnerabilities.

Sometimes the line between what counts as a change to an existing feature v.s a new feature is blurry v.s a bug fix is blurry. But that's fine.We can pick whichever we think would be most useful for readers.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,35 +1,50 @@
...
+
+### Fixed
+- [#10777](https://github.com/MetaMask/metamask-extension/pull/10777): Display BNB token image for default currency on BSC network home screen
 - [#10650](https://github.com/MetaMask/metamask-extension/pull/10650): Fix: ETH now only appears once in the swaps "to" and "from" dropdowns.
 
 ## [9.2.1] - 2021-03-26
+### Fixed
 - [#10692](https://github.com/MetaMask/metamask-extension/pull/10692): Prevent UI crash when a 'wallet_requestPermissions" confirmation is queued behind a "wallet_addEthereumChain" confirmation
 - [#10712](https://github.com/MetaMask/metamask-extension/pull/10712): Fix infinite spinner when request for token symbol fails while attempting an approve transaction
+- [#10486](https://github.com/MetaMask/metamask-extension/pull/10486): Add setting to hide zero balance tokens
```
Comment (Gudahtt) [inline CHANGELOG.md:22]: What is this doing here :thinking: This was supposed to be in v9.2.0

Comment (Gudahtt) [reply CHANGELOG.md]: Fixed in a7e6e05b90a96f8d89e3f1aa9f9b40b4fa45bb55

**Review by brad-decker — APPROVED** [review]

**Pull request merged**